jnigen/test/summary_test.dart

```diff
@@ -0,0 +1,324 @@
+    expect(listType.params, hasLength(1));
+    final tType = listType.params[0];
+    expect(tType.kind, Kind.typeVariable);
+    expect((tType.type as TypeVar).name, equals('T'));
+
+    final wm = m.params[5];
+    expect(wm.type.kind, equals(Kind.declared));
+    final mapType = (wm.type.type as DeclaredType);
+    expect(mapType.binaryName, equals('java.util.Map'));
+    expect(mapType.params, hasLength(2));
+    final strType = mapType.params[0];
+    expect(strType.name, 'java.lang.String');
+    // TODO(#141): Wildcard implementation.
+    /*
+    final wildcardType = mapType.params[1];
+    expect(wildcardType.kind, equals(Kind.wildcard));
+    expect((wildcardType.type as Wildcard).extendsBound?.name,
+        equals('java.lang.CharSequence'));
+    */
+  });
+
+  test('superclass', () {
+    final baseClass = classes.getClass('inheritance', 'BaseClass');
+    expect(baseClass.typeParams, hasLength(1));
+    final tp1 = baseClass.typeParams[0];
+    expect(tp1.bounds.map((b) => b.name).toList(), ['java.lang.CharSequence']);
+
+    final specific = classes.getClass('inheritance', 'SpecificDerivedClass');
+    expect(specific.typeParams, hasLength(0));
+    expect(specific.superclass, isNotNull);
+    final sAnc = specific.superclass!.type as DeclaredType;
+    expect(sAnc.params[0].type, isA<DeclaredType>());
+    expect(sAnc.params[0].type.name, equals('java.lang.String'));
+
+    final generic = classes.getClass('inheritance', 'GenericDerivedClass');
+    expect(generic.typeParams, hasLength(1));
+    expect(generic.typeParams[0].name, equals('T'));
+    expect(generic.typeParams[0].bounds.map((b) => b.name).toList(),
+        ['java.lang.CharSequence']);
+    expect(generic.superclass, isNotNull);
+    final gAnc = generic.superclass!.type as DeclaredType;
+    expect(gAnc.params[0].type, isA<TypeVar>());
+    expect(gAnc.params[0].type.name, equals('T'));
+  });
+
+  test('constructor is included', () {
+    final example = classes.getExampleClass();
+    void assertOneCtorExistsWithArity(int arity, List<String> paramTypes) {
+      final arityCtors = example.methods
+          .where((m) => m.name == '<init>' && m.params.length == arity)
+          .toList();
+      expect(arityCtors, hasLength(1));
+      final ctor = arityCtors[0];
+      expect(ctor.params.map((p) => p.type.name), equals(paramTypes));
+    }
+
+    assertOneCtorExistsWithArity(0, []);
+    assertOneCtorExistsWithArity(1, ['int']);
+    assertOneCtorExistsWithArity(2, ['int', 'boolean']);
+    assertOneCtorExistsWithArity(3, ['int', 'boolean', 'java.lang.String']);
+  });
+
+  test('Overloaded methods', () {
+    final methods = classes
+        .getExampleClass()
+        .methods
+        .where((m) => m.name == 'overloaded')
+        .toList();
+    expect(methods, hasLength(5));
+    final signatures =
+        methods.map((m) => m.params.map((p) => p.type.name).toList()).toList();
+    expect(
+        signatures,
+        containsAll(const [
```

Comment:
The set of signatures should not only `containAll` but also be equal to the provided signatures, you can do something similar to what you did in the enum test.